lib/_http_incoming.js

```diff
@@ -63,6 +60,13 @@ util.inherits(IncomingMessage, Stream.Readable);
 
 exports.IncomingMessage = IncomingMessage;
 
+Object.defineProperty(IncomingMessage.prototype, 'client', {
+  configurable: true,
+  enumerable: true,
+  get: util.deprecate(function() {
+    return this.socket;
+  }, 'client is deprecated, use socket or connection instead')
+});
```

Comment:
What would the setter do? Throw a deprecation error? Set `this.socket`? Set `this.client` again? Something else?